File: java/src/me/army8735/jssc/util/Character.java
package me.army8735.jssc.util;

public class Character {
	public static final char LINE = '\n';
	public static final char ENTER = '\r';
	public static final char BLANK = ' ';
	public static final char TAB = '\t';
	public static final char UNDERLINE = '_';
	public static final char DOLLAR = '$';
	public static final char SHARP = '#';
	public static final char MINUS = '-';
	public static final char AT = '@';
	public static final char SLASH = '/';
	public static final char BACK_SLASH = '\\';
	public static final char DECIMAL = '.';
	public static final char LEFT_BRACKET = '[';
	public static final char RIGHT_BRACKET = ']';
	public static final char STAR = '*';
	public static final char LEFT_PARENTHESE = '(';
	public static final char RIGHT_PARENTHESE = ')';
	public static final char COMMA = ',';
	public static final char SEMICOLON = ';';
	public static final char EQUAL = '=';
	public static final boolean isDigit(char c) {
		return c >= '0' && c <= '9';
	}
	public static final boolean isDigit16(char c) {
		return isDigit(c) || (c >= 'a' && c <= 'f') || (c >= 'A' && c <= 'F');
	}
	public static final boolean isDigit2(char c) {
		return c == '0' || c == '1';
	}
	public static final boolean isDigit8(char c) {
		return c >= '0' && c <= '7';
	}
	public static final boolean isLetter(char c) {
		return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z');
	}
	public static final int count(String s, char c) {
		int count = 0;
		int i = -1;
		while((i = s.indexOf(c, i + 1)) != -1) {
			count++;
		}
		return count;
	}
}

File: web/util/character.js
define(function(require, exports, module) {
  exports.LINE = '\n';
  exports.ENTER = '\r';
  exports.BLANK = ' ';
  exports.TAB = '\t';
  exports.UNDERLINE = '_';
  exports.DOLLAR = '$';
  exports.SHARP = '#';
  exports.MINUS = '-';
  exports.AT = '@';
  exports.SLASH = '/';
  exports.BACK_SLASH = '\\';
  exports.DECIMAL = '.';
  exports.LEFT_BRACKET = '[';
  exports.RIGHT_BRACKET = ']';
  exports.STAR = '*';
  exports.LEFT_PARENTHESE = '(';
  exports.RIGHT_PARENTHESE = ')';
  exports.COMMA = ',';
  exports.SEMICOLON = ';';
  exports.EQUAL = '=';
  exports.isDigit = function(c) {
    return c >= '0' && c <= '9';
  };
  exports.isDigit16 = function(c) {
    return exports.isDigit(c) || (c >= 'a' && c <= 'f') || (c >= 'A' && c <= 'F');
  };
  exports.isDigit2 = function(c) {
    return c == '0' || c == '1';
  };
  exports.isDigit8 = function(c) {
    return c >= '0' && c <= '7';
  };
  exports.isLetter = function(c) {
    return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z');
  };
  exports.count = function(s, c) {
    var count = 0,
      i = -1;
    while((i = s.indexOf(c, i + 1)) != -1) {
      count++;
    }
    return count;
  };
  exports.isUndefined = function(s) {
    return typeof s == 'undefined';
  };
  exports.isString = function(s) {
    return Object.prototype.toString.call(s) == "[object String]";
  };
  exports.isNumber = function(s) {
    return Object.prototype.toString.call(s) == "[object Number]";
  };
});